client/containers/CommunityCreate/CommunityCreate.js

```diff
@@ -1,177 +1,144 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
+import React, { useState } from 'react';
 import { Button, NonIdealState } from '@blueprintjs/core';
-import { ColorInput, GridWrapper, InputField, ImageUpload, PageWrapper } from 'components';
-import { hydrateWrapper, apiFetch, slugifyString } from 'utils';
+import { ColorInput, GridWrapper, InputField, ImageUpload } from 'components';
+import { apiFetch, slugifyString } from 'utils';
+import { usePageContext } from 'utils/hooks';
 
 require('./communityCreate.scss');
 
-const propTypes = {
-	communityData: PropTypes.object.isRequired,
-	loginData: PropTypes.object.isRequired,
-	locationData: PropTypes.object.isRequired,
-};
-
-class CommunityCreate extends Component {
-	constructor(props) {
-		super(props);
-		this.state = {
-			subdomain: '',
-			title: '',
-			description: '',
-			heroLogo: '',
-			accentColorDark: '#2D2E2F',
-			accentColorLight: '#FFFFFF',
-			createIsLoading: false,
-			createError: undefined,
-		};
-		this.onCreateSubmit = this.onCreateSubmit.bind(this);
-		this.onSubdomainChange = this.onSubdomainChange.bind(this);
-		this.onTitleChange = this.onTitleChange.bind(this);
-		this.onDescriptionChange = this.onDescriptionChange.bind(this);
-		this.onHeroHeaderLogoChange = this.onHeroHeaderLogoChange.bind(this);
-	}
+const CommunityCreate = () => {
+	const { loginData } = usePageContext();
+	const [subdomain, subdomainSetter] = useState('');
```

Comment:
I think we should stick with the `[val, setVal]` convention for `useState` bindings here.

Reply:
Ya, you're right. I did this because it was easier to handle the parallel conversion of 10+ state fields in components like `<UserCreate />`. Having to change the capitalization of the `v` in `[val, setVal]` always took a bunch of manual editing, but duh - there's a shortcut to do that, so my original concern is irrelevant.
```
cmd + KU    Transform to Uppercase
cmd + KL    Transform to Lowercase
```